File: fixtures/scale/python/large/cipherlab/checks/check_emberstream.py
from cipherlab.ciphers.emberstream import emberstream_encrypt, emberstream_decrypt
from cipherlab.specs.emberstream_spec import emberstream_spec
from cipherlab.corpus.corpus_stream import corpus_stream


def check_emberstream():
    spec = emberstream_spec()
    for text in corpus_stream():
        sealed = emberstream_encrypt(text, spec["key"])
        opened = emberstream_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "stream"

File: fixtures/scale/ruby/large/lib/check_emberstream.rb
require_relative 'emberstream_encrypt'
require_relative 'emberstream_decrypt'
require_relative 'corpus_stream'

def check_emberstream
  corpus_stream.each do |text|
    sealed = emberstream_encrypt(text, 13)
    opened = emberstream_decrypt(sealed, 13)
    return false if opened != text
  end
  true
end
